timrlink.net.Core/Service/TaskService.cs

```diff
@@ -151,19 +153,28 @@ public async Task SynchronizeTasksByExternalId(IDictionary<string, API.Task> exi
             }
         }
 
-        protected async Task AddOrUpdateTask(IDictionary<string, API.Task> existingTaskIDs, API.Task task, bool updateTask, IEqualityComparer<API.Task> equalityComparer)
+        protected async Task AddOrUpdateTask(IDictionary<string, API.Task> existingTaskIDs, IDictionary<string, API.Task> addedTasks, API.Task task, bool updateTask, IEqualityComparer<API.Task> equalityComparer)
         {
             logger.LogDebug($"Checking Task(Name={task.name}, ExternalId={task.externalId})");
 
-            if (existingTaskIDs.TryGetValue(task.externalId, out var existingTask))
+            if (existingTaskIDs.TryGetValue(task.externalId, out var existingTask) && updateTask)
```

Comment:
the 2 topmost if statements could be "or"-ed together, body is the same

or we differentiate in the log message, wouldn't be bad either